streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -278,14 +279,17 @@ public void flush() {
 
     @Override
     public void close() {
-        try {
-            flush();
-        } finally {
-            try {
-                super.close();
-            } finally {
-                cache.close(cacheName);
+        final List<RuntimeException> suppressed = new ArrayList<>();
+        captureException(this::flush, suppressed);
```

Comment:
As we discussed offline, I think here we can just call `cache.flush(cacheName)` instead of `this.flush` right?

Reply:
See my comment https://github.com/apache/kafka/pull/8368#discussion_r398919022